spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestUpdate.java

```diff
@@ -219,7 +220,14 @@ public void testUpdateWithoutCondition() {
     Assert.assertEquals("Should have 4 snapshots", 4, Iterables.size(table.snapshots()));
 
     Snapshot currentSnapshot = table.currentSnapshot();
-    validateCopyOnWrite(currentSnapshot, "2", "3", "3");
+    Assert.assertEquals("Operation must match", OVERWRITE, currentSnapshot.operation());
+    Assert.assertEquals("Changed partitions count must match",
+        "2", currentSnapshot.summary().get("changed-partition-count"));
+    Assert.assertEquals("Number of added data files should be same as deleted",
```

Comment:
I am afraid this is not correct. Earlier in the test, we had 3 separate insert statements that always produce 3 files. However, the number of output files is non-deterministic and can be either 2 or 3 (see my comment in the issue).

One way to fix this is to offer methods like this:

```
protected void validateProperty(Snapshot snapshot, String property, String allowedValue) {
  ...
}

protected void validateProperty(Snapshot snapshot, String property, Set<String> allowedValues) {
  ...
}
```
And then call it like this:

```
Assert.assertEquals("Operation must match", OVERWRITE, currentSnapshot.operation());
validateProperty(currentSnapshot, CHANGED_PARTITION_COUNT_PROP, "2");
validateProperty(currentSnapshot, DELETED_FILES_PROP, "3");
validateProperty(currentSnapshot, ADDED_FILES_PROP, ImmutableSet.of("2", "3"));
```

We can then refactor `validateCopyOnWrite` and `validateMergeOnRead` to use these methods.

Reply:
I should have used constants defined in `SnapshotSummary` when I wrote this part. If we plan to modify it, it would be great to switch to using constants.

Reply:
Thanks , you are right.  Fixed the checks, and also used SnapshotSummary.

But I'm curious, is it still possible to get only 1 data file in extreme worse case?

Reply:
Yeah but I doubt it is going to actually happen.